src/libraries/System.Security.Claims/src/System/Security/Claims/Claim.cs

```diff
@@ -514,17 +514,17 @@ protected virtual void WriteTo(BinaryWriter writer, byte[] userData)
 
             if ((mask & SerializationMask.HasProperties) == SerializationMask.HasProperties)
             {
-                writer.Write(_properties.Count);
+                writer.Write(_properties!.Count);
                 foreach (var kvp in _properties)
                 {
                     writer.Write(kvp.Key);
-                    writer.Write(kvp.Value);
+                    writer.Write(kvp.Value!);
```

Comment:
This write being non-nullable suggests that the Properties property should say that the value is non-null strings... because at least this method is likely to throw if a property has a null value.

Reply:
True i should have asserted here, but somehow I was not able to use `Debug.Assert(...)` even imported `System.Diagnostics;`. As you suggested it can be non null, and it still need a bang here https://github.com/dotnet/runtime/blob/148638ca5be8acbc865ecefb988be990bf85bc7e/src/libraries/System.Security.Claims/src/System/Security/Claims/Claim.cs#L276 , anyways i will make it non null, that seems better